### PR TITLE
Use hass-tabs-subpage in add-on view

### PR DESCRIPTION
**Pull request by ludeeus**
# Proposed change

Change add-on view to be tab based (by using hass-tabs-subpage), this will make room for the documentation endpoint in the backend.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] ? New feature (thank you!) 
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by pvizeli** [comment]
Any news on that? I would otherwise create a new branch "supervisor" we merge this into that so we can play with in on RL.

**Comment by ludeeus** [comment]
Didn't notice there were conflicts here.
I'll look at those tomorrow

**Comment by ludeeus** [comment]
Should be a good branch again now

**Review by bramkragten — APPROVED** [review]

File: hassio/src/addon-view/hassio-addon-router.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { HomeAssistant } from "../../../src/types";
+// Don't codesplit the others, because it breaks the UI when pushed to a Pi
+import "./info/hassio-addon-info-tab";
+import "./config/hassio-addon-config-tab";
+import "./log/hassio-addon-log-tab";
+import { HassioAddonDetails } from "../../../src/data/hassio/addon";
+
+@customElement("hassio-addon-router")
+class HassioAddonRouter extends HassRouterPage {
+  @property({ type: Boolean, reflect: true }) public narrow = false;
```
Comment (bramkragten) [inline hassio/src/addon-view/hassio-addon-router.ts:15]: Why do we reflect this?

File: hassio/src/addon-view/info/hassio-addon-info-tab.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { HomeAssistant } from "../../../../src/types";
+import { HassioAddonDetails } from "../../../../src/data/hassio/addon";
+import { hassioStyle } from "../../resources/hassio-style";
+import { haStyle } from "../../../../src/resources/styles";
+
+import "./hassio-addon-info";
+
+@customElement("hassio-addon-info-tab")
+class HassioAddonInfoDashboard extends LitElement {
+  @property({ type: Boolean, reflect: true }) public narrow!: boolean;
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info-tab.ts:19]: Why do we reflect?

File: hassio/src/addon-view/hassio-addon-router.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/addon-view/hassio-addon-router.ts:15]: Hmm.... I noticed that as well, I can not remember adding it, I just assumed it came with the rebase

Comment (ludeeus) [reply hassio/src/addon-view/hassio-addon-router.ts]: It did not, removed

File: hassio/src/addon-view/info/hassio-addon-info-tab.ts (hunk 2 above)
Comment (ludeeus) [reply hassio/src/addon-view/info/hassio-addon-info-tab.ts:19]: removed

File: hassio/src/addon-view/info/hassio-addon-info.ts [hunk 3]
```diff
@@ -91,9 +91,11 @@ const PERMIS_DESC = {
 
 @customElement("hassio-addon-info")
 class HassioAddonInfo extends LitElement {
-  @property() public hass!: HomeAssistant;
+  @property({ type: Boolean, reflect: true }) public narrow!: boolean;
```
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info.ts:94]: reflect

File: hassio/src/addon-view/info/hassio-addon-info-tab.ts (hunk 2 above)
Comment (bramkragten) [inline hassio/src/addon-view/info/hassio-addon-info-tab.ts:19]: Not removed

File: hassio/src/addon-view/config/hassio-addon-config-tab.ts [hunk 4]
```diff
@@ -58,7 +59,11 @@ class HassioAddonConfigDashboard extends LitElement {
       haStyle,
       hassioStyle,
       css`
-        .content {
+        @media screen and (min-width: 1024px) {
+          .content {
+            width: 50%;
+            margin-left: 25%;
```
Comment (bramkragten) [inline hassio/src/addon-view/config/hassio-addon-config-tab.ts:65]: You want it centered right?
```suggestion
            margin: auto;
```

Comment (ludeeus) [reply hassio/src/addon-view/config/hassio-addon-config-tab.ts]: We sure do :)

**Pull request merged**